proposals/lip-0047.md

```diff
@@ -511,21 +647,29 @@ chainProperties.roundEndHeight = g.header.height + len(snapshotStore(0).addresse
 BFTThreshold = snapshotStore(0).threshold
 setBFTParameters(BFTThreshold, BFTThreshold, snapshotStore(0).validators)
 
+# Pass the BLS keys and generator keys to the Validators module
+for every validator element in genesisBlockAssetObject.validatorSubstore.validators:
+    registerValidatorKeys(validator.address, validator.proofOfPossession, validator.generatorKey, validator.blsKey)
+
 # Pass the list of validators to the Validators module
 for i in range(len(snapshotStore(0).validators)):
     addresses = snapshotStore(0).address[i]
```

Comment:
```suggestion
    addresses = snapshotStore(0).validators[i].address
```

Reply:
the same for line 691

Reply:
Note that we also always overwrite the value `addresses`, but we actually want to append it to a list, right?

Reply:
can be resolved

Reply:
can be resolved